src/main/java/com/diamondfire/helpbot/bot/command/argument/impl/types/Argument.java

```diff
@@ -13,9 +13,14 @@
  *
  * @param <T> parsed value type
  */
-public interface Argument<T> {
+public abstract class Argument<T> {
+    
+    public T parsed(@NotNull Deque<String> args, CommandEvent event) throws ArgumentException {
+        return parseValue(args, event);
+    }
+    
+    protected abstract T parseValue(@NotNull Deque<String> args, CommandEvent event) throws ArgumentException;
     
-    T parseValue(@NotNull Deque<String> args, CommandEvent event) throws ArgumentException;
     
 }
```

Comment:
This can still stay as an interface, just you would use a default method instead. I am also not sure what the point of this is?

Reply:
FallbackArgument overrides the parseValue method

Reply:
And sure ill make it an interface with a default method